### DISCUSSION TITLE
Failed to parse source map after updating to version 4.2.5

### DISCUSSION THREAD

**Discussion by luane-aquino** [remarkablemark/html-react-parser#1094] [Q&A] [answered]
Hi guys! I updated from version `1.2.1` to `4.2.5`, but when I run the app (it is a react app create with CRA), it started to show the following warnings in the terminal, does anyone know what it causing these warning and how to solve it? thank you!

<img width="1142" alt="image" src="https://github.com/remarkablemark/html-react-parser/assets/10707843/73dd1a32-4857-4e02-a547-daf09559df5b">

**Comment by remarkablemark** [comment]
@luane-aquino can you provide a reproducible example? I see no issues with the [JavaScript](https://codesandbox.io/s/940pov1l4w) and [TypeScript CodeSandbox](https://codesandbox.io/s/html-react-parser-z0kp6) after upgrading to the latest version.

**Reply by luane-aquino** [reply]
Hi @remarkablemark ! do you know how can I reproduce? becase it is a repository of the team I work, it is a big one, with many dependencies. We use typescript version `^4.7.3` and react/react-dom version `^17.0.1`. Do you think a reproducible example with these  typescript and react version ok?

**Reply by remarkablemark** [reply]
Yes you can try forking the CodeSandbox links above and updating the dependencies to match your project. If you get the same warnings, then I can take a look at it.

**Reply by luane-aquino** [reply]
Hi @remarkablemark , I could not reproduce it https://codesandbox.io/s/html-react-parser-forked-ppr4yh?file=/src/index.tsx. But I have investigated more and it seems a problem with `html-dom-parser` sourcemap, but I dont't know how to solve it. 

As the error in the terminal says, it can't find `constants.ts` file, I looked at  my `node_modules`, and indeed there is not such file, there is a `constants.js` file in another path though.
print of my node modules:
<img width="462" alt="image" src="https://github.com/remarkablemark/html-react-parser/assets/10707843/fd73a07c-1106-4bc5-bf59-27311064d5ca">

Besides that, I tried cleaning the cache `yarn cache clean` but didn't work.

**Reply by remarkablemark** [reply]
This is related to `html-dom-parser`'s migration to TypeScript in https://github.com/remarkablemark/html-dom-parser/pull/527

The fix should be to include `src` files as part of the publish step or enable `inlineSources` in tsconfig. ~~Can I get your help in creating a bug issue?~~

Created #1097

**Reply by remarkablemark** [reply]
This will be fixed in #1098

**Reply by remarkablemark** [reply, marked as answer]
Release:

- https://github.com/remarkablemark/html-react-parser/pull/1099#issuecomment-1771962965

**Comment by luane-aquino** [comment]
Hi @remarkablemark sorry for the late reply 🥲, and also thanks a lot for creating the issue for me and for fixing it 🎉🙏

**Reply by remarkablemark** [reply]
Thanks for the update!
